src/util.ts

```diff
@@ -12,18 +12,32 @@ export function minBy<T>(
   let minItem = array[minIndex]
   let minValue = value(minItem)
 
-  array.slice(1).forEach((item, index) => {
-    const currentValue = value(item)
+  range(1, array.length).map((index) => {
+    const currentValue = value(array[index])
     if (currentValue < minValue) {
       minValue = currentValue
-      minItem = item
+      minItem = array[index]
       minIndex = index
     }
   })
 
   return [minItem, minIndex]
 }
 
+export function pairs<T>(array: T[]): [T, T][] {
```

Comment:
Not sure about abstracting this function to here when it is only used in geometry

Reply:
good point! I'll move it there

Reply:
done :)